src/modules/ttlmonitor/TTLMonitorBase.cpp

```diff
@@ -188,15 +185,17 @@ void TTLMonitorBase::processing_thread() {
       usleep(10);
     }
 
-    RuntimeStatusMask = RtStat.getRuntimeStatusMask(
-        {ITCounters.RxPackets, Counters.MonitorCounts, Counters.TxBytes});
-    for (auto &serializer : Serializers) {
-      if (serializer.ProduceTimer.timeout()) {
-        XTRACE(DATA, DEB, "Serializer timed out, producing message now");
-        Counters.TxBytes += serializer.produce();
+    if (ProduceTimer.timeout()) {
+      RuntimeStatusMask = RtStat.getRuntimeStatusMask(
+          {ITCounters.RxPackets, Counters.MonitorCounts, Counters.TxBytes});
+      for (auto &serializer : Serializers) {
```

Comment:
I'm not sure if double stacking the timer checks is a good idea, we should use one universal produce timer, or each serializer's own timer, but not both